Add linter to packages/ folder

rafeca opened the pull request:
### Requirements for Contributing a Bug Fix

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix an existing bug. To contribute other changes, you must use a different template. You can see all templates at https://github.com/atom/atom/tree/master/.github/PULL_REQUEST_TEMPLATE.
* The pull request must update the test suite to demonstrate the changed functionality. For guidance, please see https://flight-manual.atom.io/hacking-atom/sections/writing-specs/.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see https://github.com/atom/atom/tree/master/CONTRIBUTING.md#pull-requests.

### Identify the Bug

Linter was not checking the `packages/` folder.

### Description of the Change

The JS code style of the `packages/` folder is not matching the style of the rest of the repository, this is caused by not having the linter running on this folder.

This PR enables the linter and fixes all the linting issues that appeared on that folder. Most of the issues have been fixed automatically by running [`prettier-standard`](https://www.npmjs.com/package/prettier-standard) on that folder, the things that needed to be fixed manually are in separate commits.

### Alternate Designs

N/A

### Possible Drawbacks

This PR changes a bunch of files, and while `prettier` should only do style changes, there's always the possibility of breaking something when making big changes. To mitigate this risk I'm gonna review some of the changes manually and make sure that they're ok.

### Verification Process

I'm going to manually verify the automatic fixes and rely on the existing automated tests.

### Release Notes

N/A (it's only an internal change)

rafeca committed:
Run prettier on packages/ folder

rafeca committed:
Fix syntax error in fixture

rafeca committed:
Remove unused variables from specs

rafeca committed:
Add curly braces around if statements

rafeca committed:
Add comment and ignore rule on regular expression.

rafeca committed:
Enable linter on packages/ folder

rafeca commented on packages/line-ending-selector/lib/main.js:
```diff
@@ -1,12 +1,16 @@
+
+// the following regular expression is executed natively via the `substring` package,
+// where `\A` corresponds to the beginning of the string.
+// eslint-disable-next-line no-useless-escape
```
@jasonrudolph, @smashwilson can you verify if what I wrote here is correct? 🙃

jasonrudolph replied:
@rafeca: That sounds mostly right. I see that the `\A` was introduced in https://github.com/atom/line-ending-selector/pull/56. The explanation in that PR might help you craft the comment above and/or you could consider including a link to that PR. Does that help?

rafeca replied:
Yes, thank you!

Arcanemagus commented on packages/incompatible-packages/lib/status-icon-component.js:
```diff
@@ -14,7 +14,7 @@ export default class StatusIconComponent {
     return (
       <div className='incompatible-packages-status inline-block text text-error'>
-        <span className='icon icon-bug'></span>
+        <span className='icon icon-bug' />
```
The ending `</span>` isn't allowed to be omitted in valid HTML, is etch fixing this or are we just relying on Chrome to fix it for us (which should be fixed)?

rafeca replied:
Good point! in this case I'm pretty sure that this code get transpiled (probably by babel) into `etch` calls to create DOM elements, since this is standard JSX.

Probably @nathansobo or @as-cii know better :)

Arcanemagus reviewed (commented):
Thanks for starting this! Always great to see more code brought under enforcement of the coding style/validity checking.

jasonrudolph reviewed (approved):
> This PR changes a bunch of files, and while `prettier` should only do style changes, there's always the possibility of breaking something when making big changes. To mitigate this risk I'm gonna review some of the changes manually and make sure that they're ok.automated tests.

I can't claim to have reviewed every line, but as noted in the PR body, "prettier should only do style changes," so this should be a safe change to merge. And since you're planning to review the non-automatic changes manually to make sure that they're OK, I think this is safe to ship once you've completed your verification.

rafeca committed:
Lint fixture file from go-to-line package

The specs needed to be updated since they depended on the actual contents
of the fixture.

rafeca committed:
Lint fixture file from git-diff package

rafeca committed:
Small linter fixes

rafeca commented:
I've reviewed the automatic changes done by `prettier` everything I found is safe:
* Whitespace changes to avoid long lines.
* Use single quotes for strings whenever it's possible.
* Remove semicolons when it's safe.
* Does not add trailing commas on objects or functions, so it should be safe on old JS engines.

The failing test is related to `watchPath`. Since that code was not changed in this PR I'm assuming that the error is not related and that that test is flaky (I've seen it failing on other builds).

smashwilson commented:
> The failing test is related to `watchPath`. Since that code was not changed in this PR I'm assuming that the error is not related and that that test is flaky (I've seen it failing on other builds).

👍 Yeah, I'm (slowly) investigating that one in #18857.
